server/src/main/java/io/druid/segment/realtime/firehose/ReplayableFirehoseFactory.java

```diff
@@ -162,6 +162,11 @@ public ReplayableFirehose(InputRowParser parser) throws IOException
               while (delegateFirehose.hasMore() && cos.getCount() < getMaxTempFileSize()) {
                 try {
                   InputRow row = delegateFirehose.nextRow();
+
+                  if (row == null) {
+                    continue;
```

Comment:
I think this should be removed because ReplayableFirehose is able to simply replay the original data without filtering any rows. Index tasks should decide that the given row will be skipped or not.